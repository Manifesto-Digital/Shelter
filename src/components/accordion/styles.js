import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
