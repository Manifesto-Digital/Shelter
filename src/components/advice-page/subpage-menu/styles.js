import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
