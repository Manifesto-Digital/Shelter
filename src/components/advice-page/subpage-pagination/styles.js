import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
