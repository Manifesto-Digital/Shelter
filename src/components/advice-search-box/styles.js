import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
