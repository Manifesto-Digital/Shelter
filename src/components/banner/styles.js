import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
