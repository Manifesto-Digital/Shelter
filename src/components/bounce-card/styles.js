import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
