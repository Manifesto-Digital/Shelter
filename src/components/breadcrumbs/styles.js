import styled from 'styled-components';
import LinkHandler from '../link-handler';

export const BreadcrumbList = styled.ul`
  margin: 0;
  padding: 0.8em 0;
`;

export const BreadcrumbListItem = styled.li`
  display: inline-block;
  list-style: none;
  margin: 0 0.4em 0 0;
  padding: 0;
  font-size: ${({ theme }) => theme.fontsize.small};

  &:after {
    content: '>';
    margin-left: 0.4em;
  }
  &:last-child {
    &:after {
      content: '';
    }
  }
`;

export const BreadcrumbAnchor = styled(LinkHandler)`
  text-decoration: none;
`;
