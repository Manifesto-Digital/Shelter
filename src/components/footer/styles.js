import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { Link } from 'gatsby';
import { breakpoint } from '../theme/breakpoint';
import { StyledSVG } from '../share-block/styles';
import LinkHandler from '../link-handler';
import Accordion from '../accordion';
import { HeadingButton, Heading, Content } from '../accordion/styles';

export const Logo = styled(SVG)`
  display: block;
  width: 150px;
`;

export const LogoWrapper = styled(Link)`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 52px;
  margin-right: auto;

  ${breakpoint.desktop`
    margin-bottom: ${({ theme }) => theme.spacing.standard};
    padding: ${({ theme }) => theme.spacing.standard} 25px;
    margin-right: 0;
    margin-bottom: 0;
  `}
`;

export const Wrapper = styled.footer`
  display: flex;
  flex-direction: column;
  background: ${({ theme }) => theme.palette.black};
  margin-top: ${({ theme, removeMarginTop }) =>
    removeMarginTop ? '0' : theme.spacing.xl};
  padding-top: ${({ theme }) => theme.spacing.standard};

  ${breakpoint.tablet`
    padding-top: ${({ theme }) => theme.spacing.xl};
  `}
`;

export const Top = styled.div`
  display: flex;

  > div {
    width: 100%;
  }
`;

export const Social = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.standard};

  ${StyledSVG} {
    color: ${({ theme }) => theme.palette.white};
    fill: ${({ theme }) => theme.palette.white};
  }
`;

export const Menus = styled.nav`
  width: 100%;
  font-size: ${({ theme }) => theme.fontsize.small};
  color: ${({ theme }) => theme.palette.white};
  margin-bottom: ${({ theme }) => theme.spacing.standard};
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  grid-gap: ${({ theme }) => theme.spacing.standard};

  @media (min-width: '641px') {
    grid-template-columns: repeat(4, 1fr);
    grid-gap: ${({ theme }) => theme.spacing.standard};
  }
`;

export const MenuList = styled.ul`
  display: flex;
  flex-direction: column;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;

  ${breakpoint.desktop`
    width: auto;
  `}
`;

export const ItemLink = styled(LinkHandler)`
  font-size: ${({ theme }) => theme.fontsize.small};
  color: ${({ theme }) => theme.palette.white};

  &:hover {
    color: ${({ theme }) => theme.palette.primary};
  }
`;

export const Item = styled.li`
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-bottom: ${({ theme }) => theme.spacing.standard};
  padding-left: 0;
`;

export const Bottom = styled.div`
  position: relative;
  padding: ${({ theme }) => theme.spacing.large} 0;
  background-color: ${({ theme }) => theme.palette.grey66};
  overflow: hidden;
`;

export const BottomInner = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  color: ${({ theme }) => theme.palette.white};
  font-size: ${({ theme }) => theme.fontsize.small};

  ${breakpoint.desktop`
    flex-direction: row;
    margin-bottom: 0;
  `}
`;

export const ContentWrapper = styled.div`
  flex: 0 1 100%;
  width: 100%;
  text-align: left;
`;

export const FooterAccordion = styled(Accordion)`
  flex: 1;
  color: ${({ theme }) => theme.palette.grey10};

  &:not(:first-of-type) {
    &:before {
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }

  h3 {
    ${breakpoint.tablet`
      margin-bottom: ${({ theme }) => theme.spacing.large};
  `};
  }

  ${Content} {
    padding-left: ${({ theme }) => theme.spacing.standard};
  }
  ${Heading} {
    color: ${({ theme, isOpen }) =>
      isOpen ? theme.palette.primary : theme.palette.white};
    ${breakpoint.tablet`
      font-size: ${({ theme }) => theme.headers.h3};
  `};
  }
  ${HeadingButton} {
    color: ${({ theme, isOpen }) =>
      isOpen ? theme.palette.primary : theme.palette.white};
    padding-left: ${({ theme }) => theme.spacing.standard};
  }
`;

export const SiteDate = styled.div`
  position: absolute;
  top: 0;
  left: 0;

  width: 37px;
  ${breakpoint.tablet`
       width: 43px;
      `};
`;

export const Text = styled.small`
  position: relative;
  width: 100%;

  div {
    &:last-child {
      text-indent: 37px;
      ${breakpoint.tablet`
       text-indent: 43px;
      `};
    }
  }
`;
