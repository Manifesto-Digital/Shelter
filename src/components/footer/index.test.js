import React from 'react';
import 'jest-styled-components';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import { createHeaderNavigation } from '../../utils/test-factories';
import { PureFooter } from './index';
import { ItemLink } from './styles';
import { hidePascalCaseWarning } from '../../utils/test-mocks';

it('Renders correctly', () => {
  const mockData = createHeaderNavigation();
  snapshotComponent(<PureFooter pageData={mockData} />);
});

hidePascalCaseWarning();

test('Should populate a menu item correctly', () => {
  const mockData = createHeaderNavigation({
    navigationItems: [
      {
        menuLabel: 'Test label',
        navigationLink: [{ slug: 'test-page', title: 'Test Page' }],
      },
    ],
  });
  const wrapper = mountWithTheme(<PureFooter pageData={mockData} />);

  expect(
    wrapper
      .find(ItemLink)
      .at(0)
      .text()
  ).toEqual(mockData.navigationItems[0].navigationLink[0].title);

  expect(
    wrapper
      .find(ItemLink)
      .at(0)
      .props().internalLink
  ).toEqual(mockData.navigationItems[0].navigationLink[0]);
});
