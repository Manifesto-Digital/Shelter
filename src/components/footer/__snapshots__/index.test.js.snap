// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
.c2 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
  padding: 0;
}

.c12 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

.c5 {
  position: relative;
}

.c5:not(:first-of-type):before {
  position: absolute;
  content: ' ';
  left: 0;
  top: 0;
  width: 100%;
}

.c7 {
  position: relative;
  font-weight: normal;
  color: #4d63b5;
}

.c16 ul {
  padding-left: 0;
  list-style: none;
}

.c16 ul li {
  position: relative;
  padding-left: 40px;
}

.c16 ul li::before {
  content: ' ';
  position: absolute;
  top: 8px;
  left: 10px;
  display: block;
  width: 10px;
  height: 10px;
  background: #212122;
}

.c16 ul li ul {
  padding-left: 0;
  list-style: none;
}

.c16 ul li ul li:before {
  background: transparent;
  border: 1px solid #212122;
}

.c16 ol li::before {
  min-width: 10px;
}

.c16 blockquote {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  margin-right: 20px;
  margin-bottom: 0;
  padding: 60px 20px 20px 20px;
  border-left: 4px solid #ea232d;
  font-size: 1.35em;
  font-weight: 300;
}

.c16 blockquote:before {
  content: '';
  display: block;
  position: absolute;
  width: 30px;
  height: 30px;
  background-image: url(test-file-stub);
  top: 10px;
  left: 20px;
  color: #ea232d;
}

.c18 {
  display: block;
  width: 150px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 52px;
  margin-right: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #212122;
  margin-top: 60px;
  padding-top: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > div {
  width: 100%;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
}

.c15 .c19 {
  color: #fff;
  fill: #fff;
}

.c3 {
  width: 100%;
  font-size: 0.775rem;
  color: #fff;
  margin-bottom: 20px;
  display: grid;
  grid-template-columns: repeat(1,1fr);
  grid-gap: 20px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c10 {
  font-size: 0.775rem;
  color: #fff;
}

.c10:hover {
  color: #ea232d;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 20px;
  padding-left: 0;
}

.c11 {
  position: relative;
  padding: 40px 0;
  background-color: #575757;
  overflow: hidden;
}

.c13 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  color: #fff;
  font-size: 0.775rem;
}

.c14 {
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
  width: 100%;
  text-align: left;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #e5e5e5;
}

.c4 .c21 {
  padding-left: 20px;
}

.c4 .c6 {
  color: #fff;
}

.c4 .c20 {
  color: #fff;
  padding-left: 20px;
}

@media (min-width:40.0625em) {
  .c2 {
    padding: 0 20px;
  }
}

@media (min-width:62em) {
  .c17 {
    margin-bottom: 20px;
    padding: 20px 25px;
    margin-right: 0;
    margin-bottom: 0;
  }
}

@media (min-width:40.0625em) {
  .c0 {
    padding-top: 60px;
  }
}

@media (min-width:40.0625em) {
  .c3 {
    grid-template-columns: repeat(4,1fr);
    grid-gap: 20px;
  }
}

@media (min-width:62em) {
  .c8 {
    width: auto;
  }
}

@media (min-width:62em) {
  .c13 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-bottom: 0;
  }
}

@media (min-width:62em) {
  .c14 {
    -webkit-flex: 0 1 50%;
    -ms-flex: 0 1 50%;
    flex: 0 1 50%;
    width: 50%;
    max-width: 340px;
  }
}

@media (min-width:40.0625em) {
  .c4 h3 {
    margin-bottom: 40px;
  }
}

@media (min-width:40.0625em) {
  .c4 .c6 {
    font-size: 1.35em;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <nav
        aria-label="Footer menu"
        className="c3"
      >
        <div
          className="c4 c5"
        >
          <h3
            className="c6 c7"
          >
            Housing Advice
          </h3>
          <ul
            className="c8"
          >
            <li
              aria-haspopup={true}
              className="c9"
              role="menuitem"
            >
              <Link
                className="c10"
                to="/test-page"
              >
                Housing Advice
              </Link>
            </li>
            <li
              className="c9"
            >
              <Link
                className="c10"
                to="/sprint-7-legal-page-demo"
              >
                Benefits
              </Link>
            </li>
          </ul>
        </div>
        <div
          className="c4 c5"
        >
          <h3
            className="c6 c7"
          >
            What we do
          </h3>
          <ul
            className="c8"
          >
            <li
              aria-haspopup={false}
              className="c9"
              role="menuitem"
            >
              <Link
                className="c10"
                to="/test-page"
              >
                What we do
              </Link>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="c11"
  >
    <div
      className="c12"
    >
      <div
        className="c13"
      >
        <div
          className="c14"
        >
          <div
            className="c15"
          />
          <div
            className="c16"
          >
            <p>
              Some body text
            </p>
            <h2>
              Some header text
            </h2>
            <p>
              <b>
                some bollld text
              </b>
            </p>
          </div>
        </div>
        <Link
          className="c17"
          to="/"
        >
          <span
            className="isvg loading c18"
          />
        </Link>
      </div>
    </div>
  </div>
</div>
`;
